common/src/main/java/fr/openent/form/core/model/Question.java

```diff
@@ -0,0 +1,128 @@
+package fr.openent.form.core.model;
+
+import fr.openent.form.core.enums.ChoiceTypes;
+import io.vertx.core.json.JsonObject;
+import fr.openent.form.core.constants.Fields;
+
+public class Question {
+    private Integer questionType;
+    private Number id;
+    private Number questionId;
+    private String value;
+    private Number position;
+    private Number nextSectionId;
+    private Number nbResponses;
+    private ChoiceTypes type;
+    private Number code;
+    private String name;
+    private Number matrixId;
+    private Number matrixPosition;
+
+    public Question() {
+    }
+
+    public Question(JsonObject question) {
```

Comment:
Magic string pour tous les champs du constructeur

Reply:
ils sont pas déjà en magic string ??

Reply:
Ce qu'on appelle magic string c'est des string qui vont toujours être constant comme ici. Et justement on préfère les éviter, dans ces cas là on préfère faire des constantes et utiliser ces constantes (comme tu l'as fais dans le toJson par exemple)

Reply:
si j'ai bien compris on aura ça ds le constructeur :
 public Question(JsonObject question) {
        this.questionType = question.getInteger("QUESTION_TYPE", null);
        this.id = question.getInteger("ID", null);
        this.name = question.getString("NAME", null);
        this.questionId = question.getNumber("QUESTION_ID", null);
        this.value = question.getString("VALUE", "");
        this.position = question.getNumber("POSITION", null);
        this.nextSectionId = question.getNumber("NEXT_SECTION_ID", null);
        this.nbResponses = question.getNumber("NB_RESPONSES", null);
        this.code = question.getNumber("CODE", null);
        this.type = ChoiceTypes.getChoiceTypes(question.getString("TYPE", null));
        this.matrixId = question.getInteger("MATRIX_ID",null);
        this.matrixPosition = question.getInteger("MATRIX_POSITION",null);